### PR TITLE
[Components - FontSizePicker]: Use incremental sequence of numbers as labels for the available font-sizes at the segmented control (conditionally)

### PR DESCRIPTION
**Pull request by ntsekouras**
Part of https://github.com/WordPress/gutenberg/issues/36545

>There are only 5 font sizes but the styles sidebar still shows a big dropdown. Make sure Twenty twenty two theme uses the segmented control for font sizes.

Currently the logic for showing the segmented control depends on having at most five font sizes and only if none of them contain css complex values (`clamp, var, etc..`). This is because on the segmented control we show as labels the actual numeric value of the font size and it's not possible to calculate these dynamic values.

To accommodate use cases where at most five font sizes are available but also contain complex css values, this PR proposes to use an incremental sequence of numbers (1/2/3/4/5) as labels for the available font-sizes  automatically. As @jasmussen noted:
>Outside of revisiting those default slugs, there's an argument to make that the numbers catch a wider range of options. For better and worse, the numbers just go from small to larger, and need to be (and are) paired with the labels the theme defines.

The previous approach with adding an `alias` property to font size object is also a convention that could create more confusion to theme developers.

https://user-images.githubusercontent.com/16275880/146148549-77a53abe-34f9-4b7b-99b8-7adb54ceabe9.mov

#### Storybook (`npm run storybook:dev`)
Change values through the available knobs in `With Complex Css Values` story

#### With a block theme
1. In a block theme you test add to your `fontSizes` at least one complex css value
2. Make sure to have at most five font sizes

### CODE REVIEW

**Comment by oandregal** [comment]
Nice! Do we want to provide a default alias for the font sizes declared in Gutenberg's (and core's) `theme.json` [here](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-5.9/theme.json#L196)? I think that'd be a good way to highlight how it can be used.

**Comment by ntsekouras** [comment]
>Do we want to provide a default alias for the font sizes declared in Gutenberg's (and core's) theme.json here? I think that'd be a good way to highlight how it can be used.

If we go with the approach to use it only when complex css values are available (see PR's description), there is no need to add `alias` there as there are only simple css values declared.

**Comment by oandregal** [comment]
> If we go with the approach to use it only when complex css values are available (see PR's description), there is no need to add alias there as there are only simple css values declared.

Oh, interesting. If the design intention is to show numbers in the segmented control, what if we only allowed numbers in the alias field? The advantage would be that the data the users see is always consistent (numbers) no matter how the theme is set up internally.

File: packages/components/src/font-size-picker/index.js
Comment (ciampo) [inline packages/components/src/font-size-picker/index.js]: Just a though — I wonder if the decision to display an alias should be made for each "font size" independently — i.e.:

- There are more than 6 presets? Display a select control
- Less than 6 presets:
  - Display a select control only if at least one font size is not a simple control, and doesn't have an alias either
  - Otherwise, display a toggle group

This mean that, as far as all "complex" values are paired with an alias, we should be able to show a toggle group

File: packages/components/src/font-size-picker/index.js
Comment (ciampo) [inline packages/components/src/font-size-picker/index.js]: What happens if a font size is a "simple" value, but has an associated alias? Should the alias take the precedence?

**Review by ciampo — COMMENTED** [review]
Hey @ntsekouras , thank you for opening this PR!

- Should we add some unit tests and a Storybook example for this scenario?
- Would you be able to also add a [CHANGELOG](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CHANGELOG.md) entry?

> Oh, interesting. If the design intention is to show numbers in the segmented control, what if we only allowed numbers in the alias field? The advantage would be that the data the users see is always consistent (numbers) no matter how the theme is set up internally.

I think that allowing numbers as aliases would create confusion, since the main reason why we need these aliases in the first place is because they need to represent "dynamic" values.

File: packages/components/src/font-size-picker/README.md
Comment (aaronrobertshaw) [inline packages/components/src/font-size-picker/README.md]: Feel free to ignore this suggestion. It's my attempt to make this read a little better, so it's completely subjective.

```suggestion
An array of font size objects. Each object should contain the properties `size`, `name`, and `slug`. Optionally, `alias` can be included for special use cases as described below.
```

File: packages/components/src/font-size-picker/index.js
Comment (aaronrobertshaw) [reply packages/components/src/font-size-picker/index.js]: Marco's suggestion here matches the preconceived notion I had of how this would work before I read the PR description.

My thinking was that the alias would act as an override or fallback. Something for theme authors to provide when we can't determine a sensible value from the font size. 

I can also see the other angle where this could lead to an inconsistent UI. A theme author adds aliases such as `XL` for complex values but omits aliases for simple ones, resulting in a mixture of numeric and string labels in the segmented control. A similar situation could arise if we wished to use the aliases for hints within the select control.

If we made the decision to display aliases at the individual font size level, it might alleviate the need to plumb through `shouldUseAliases`.

**Review by aaronrobertshaw — COMMENTED** [review]
Very nice @ntsekouras ✨ 

This tests as advertised for me in both the Storybook and Block Editor. The knobs in the story to automate updating the font size configuration are a nice touch, thanks 👍.

In general, the `alias` approach looks like a good step forward and might only need a few tweaks as proposed by others.

**Comment by oandregal** [comment]
I'm preparing the corresponding PR for wp-cli (i18n command) at https://github.com/wp-cli/i18n-command/pull/292

**Comment by oandregal** [comment]
> I think that allowing numbers as aliases would create confusion, since the main reason why we need these aliases in the first place is because they need to represent "dynamic" values.

I see. I'm still trying to wrap my head around the design direction for this component, so bear with me :) The thought I'm struggling with is that this component displays data differently depending on the theme's internal implementation: sometimes numbers and sometimes letters. It'd be great if we could find a way to show the same type of data. Of course, not a blocker for this PR.

**Comment by oandregal** [comment]
If we can land this one quick and then polish the details it'd be helpful to help us land the corresponding wp-cli PR https://github.com/wp-cli/i18n-command/pull/292 (we shouldn't merge that one until this direction is confirmed and merged).

**Comment by jasmussen** [comment]
Thanks for the PR! I'll share some more high level thoughts, that can hopefully be helpful. 

What we want to avoid is this:

presets

With only 5 sizes defined, a segmented control would be a much more immediate and convenient control.

Going back to [this design](https://github.com/WordPress/gutenberg/issues/34345#issuecomment-926577197) and [this clarification](https://github.com/WordPress/gutenberg/issues/36545#issuecomment-972248030), it seems like the best path forward is:

1. If we have basic values (px), show numbers.
2. If we don't have basic values, show preset size values, S/M/L/XL/XXL.

2 is where it gets difficult. Here's what TT2 offers:

```
{
	"name": "Small",
	"size": "1rem",
	"slug": "small"
},
{
	"name": "Normal",
	"size": "1.125rem",
	"slug": "normal"
},
{
	"name": "Medium",
	"size": "1.75rem",
	"slug": "medium"
},
{
	"name": "Large",
	"size": "clamp(1.75rem, 3vw, 2.25rem)",
	"slug": "large"
},
{
	"name": "Huge",
	"size": "clamp(2.5rem, 4vw, 3rem)",
	"slug": "huge"
}
```

On the one hand TT2 might be bordering on _too clever_ with the clamp based values, at the same time: themes can do this, so they will. But we still can't extract numbers from that, at best we'd end up here:

```
1   1.125    1.75    2.25    3
```

Those aren't helpful indicators, and might even wrap onto multiple lines. 

Question: can we try to force T-shirt sizes, S/M/L/XL/XXL, in sequence? First defined value is "S", 5th define value is "XXL"? 

Adding the explicit alias feels like letting the genie out of the bottle, allowing theme authors to add long aliases that cause a segmented control to wrap. Ultimately it might be what we need to do, but it would be nice to first see if the predefined values could work. What do you think?

File: packages/components/src/font-size-picker/index.js
Comment (ntsekouras) [reply packages/components/src/font-size-picker/index.js]: >I wonder if the decision to display an alias should be made for each "font size" independently

I agree with Aaron:
>where this could lead to an inconsistent UI. A theme author adds aliases such as XL for complex values but omits aliases for simple ones, resulting in a mixture of numeric and string labels in the segmented control.

File: packages/components/src/font-size-picker/index.js
Comment (ntsekouras) [reply packages/components/src/font-size-picker/index.js]: Based on the initial design with the intention of showing numeric values, I'd say no..

**Comment by ntsekouras** [comment]
Thanks all the reviews and help here! ❤️ 


>Should we add some unit tests and a Storybook example for this scenario?

@ciampo there is a story for this. Regarding the unit tests, did you have something specific in mind like for taking a snapshot of the rendered html?

>Would you be able to also add a CHANGELOG entry?

That will be added yes.


>Question: can we try to force T-shirt sizes, S/M/L/XL/XXL, in sequence? First defined value is "S", 5th define value is "XXL"?

Probably no, because even if this should be the right declaration order (smaller->bigger) this is up to the theme and that order is not guaranteed. 

>Adding the explicit alias feels like letting the genie out of the bottle, allowing theme authors to add long aliases that cause a segmented control to wrap. Ultimately it might be what we need to do, but it would be nice to first see if the predefined values could work. What do you think?

Actually there could be a simpler alternative, not so flexible but then again this PR accommodates more complex use cases from theme developers, who know what they are doing, and also handles the `twenty twenty two` use case. The approach would be:
1. If we have at most five font sizes (like now)
2. If there is at least on complex css value
3. Extract the `first letter` of the font sizes (with any necessary extra small checks like no duplicates etc..)

**Pros:**
1. No extra property that is used conditionally based on the implementation details
2. Reduce confusion about when/where the `alias` should be added

**Cons:**
1. Not being able to explicitly set an alias to achieve something like this `huge->XL` (IMO not a big deal).

I think it's probably better to loose the `alias` after all, and either go with a first letter(s) approach or even the `T-shirt` sizes approach from Joen.

What do you think?

**Comment by jasmussen** [comment]
> Extract the first letter of the font sizes (with any necessary extra small checks like no duplicates etc..)

So if I had Small/Normal/Medium/Large/Huge it'd be something like this?

```
S   N   M   L   H
```

While I like the idea of starting with simple automation before adding any additional JSON properties, I can't help but feel like those letters don't translate well to _sizes_ in the same way the T-shirt labels do. I understand that a fixed sequence isn't going to be accurate if a developer decides to start with Large as the first one, or even mix and match. That that doesn't seem like a use case to optimize for.

This is not a very strong opinion, but to keep things simple, I'd start with numbers and t-shirt sizes, and then revisit the t-shirt sizes if the fixed sequence turns out to be a problem in practice.

**Comment by kjellr** [comment]
For the case of Twenty Twenty-Two, we can change those names to something else if necessary — we were just borrowing the names from Gutenberg when we named those.

**Comment by ntsekouras** [comment]
I updated the PR's description with the `T-shirt` sizes approach.

**Comment by jasmussen** [comment]
Thanks for all the work and conversation on this PR. It really shows there's always good challenges to find if you go deep on something. That's good, and I appreciate all the thoughtful feedback.

Going back to the source, #36545, the challenge is outlined in only few words:

> - There are only 5-6 font sizes but the styles sidebar still shows a big dropdown. Make sure Twenty twenty two theme uses the segmented control for font sizes.

TwentyTwentyTwo is specifically noted because it's a default theme and already now, through all its themes and patterns, show how well the block editor can be leveraged to create extremely flexible themes. It's my favorite ever, because of that, and it's beautiful.

It also uses `clamp` for fluid typography in sizes "Large" and "Huge", which is what is causing the dropdown to appear:

> - - Any complex CSS values currently force a dropdown

Given the focus on a great default out of box experience, we might want to let that inform the extent of the solution. With that in mind, we could do two things:

**1. Remove the clamp values from TwentyTwentyTwo, fall back to simple sizes**

The end result is not bad:

tt2

Fluid typography is something I'd love to see the block editor itself handle the complexity of, to the point that it could just be a set and forget toggle somewhere which then extrapolates [the math necessary to scale type well](https://codepen.io/joen/pen/eYRQyPZ).

Going this way would also let themes that really do need complex clamp values keep the dropdown, where the added space affords more descriptive labels.

**2. Go with a T-shirt size sequence**

If the clamp values are crucial for TwentyTwentyTwo's large and huge headings then I think we should go with the T-shirt sizes, as going back to the original mockups, that feels like the design that was created for the segmented control when number values aren't appropriate. That seems to be what the PR does a the moment (thank you @ntsekouras). 

To me, the benefits (intuitive labels, no need to add aliases) outweight the downsides (sequence is always from small to large). If we did this, we'd probably want to rename the verbose font sizes in TT2 anyway to match (medium instead of normal, etc.)

I don't personally have a strong preference between the two. But I do think it's important we keep the solution simple, since the original ticket focuses primarily on the TwentyTwentyTwo default experience. What do you think?

**Review by jasmussen — APPROVED** [review]
What I see in the branch as is, is this:
sizes

That is — when there are only 5 font sizes, the t-shirt monikers are applied in descending order. This feels intuitive and obvious, and it just works without needing changes to themes (other than if a theme's font sizes are not in descending order, which I'd consider an ill-advised edgecase anyway). 

Most notably, this doesn't introduce any new APIs, which makes it rather safe to try. That would also let us revisit this without deprecations if in the future we find that the t-shirt sizes don't work as intuitively as we had hoped. 

Because of that, I think we should move forward with this one, because we can always revisit. Thanks for the work!

**Comment by pablohoneyhoney** [comment]
Smaller note, but is the tooltip too attached in this context?

 image

File: packages/components/CHANGELOG.md
Comment (ciampo) [inline packages/components/CHANGELOG.md]: typo

```suggestion
-   Show a `T-shirt size` alias (S/M/L/XL/XXL) as a label of the font size, when we have at most five font sizes, where at least one the them contains a complex css value(clamp, var, etc..). We do this because complex css values cannot be calculated properly and the `T-shirt size` alias can help the user better mentally map the different available font sizes. ([#37038](https://github.com/WordPress/gutenberg/pull/37038))
```

**Review by ciampo — COMMENTED** [review]
Thank you @ntsekouras for addressing feedback, and sorry for the delay in replying!

> Regarding the unit tests, did you have something specific in mind like for taking a snapshot of the rendered html?

I was thinking about some tests about the logic to display select vs togglegroup, and using aliases or not, for example:

- check that if there are more than 6 font sizes, a select is rendered
- check that if there are 5 or less font size, a toggle group is rendered
- check that when a toggle group is rendered with only simple values, the actual values are used in the UI
- check that when a toggle group is rendered with at least one complex value, the t-shirt sizes are used

> Smaller note, but is the tooltip too attached in this context?

I agree about the tooltip positioning, but I think it should be addressed in a separate PR

**Comment by mtias** [comment]
Regarding tooltip positioning: https://github.com/WordPress/gutenberg/issues/37076

**Comment by ciampo** [comment]
Hey @ntsekouras , the unit tests that you added look good to me — thank you!

I can spot some other failing unit tests, but otherwise for me this PR is also good to go 🚀

**Comment by kjellr** [comment]
> Remove the clamp values from TwentyTwentyTwo, fall back to simple sizes

Yeah, I'm only ok doing this if we do have a fallback in place. Twenty Twenty-Two uses some pretty large sizes, and I'd like to be sure they scale down nicely. Besides, I think the unit-less values we'd be falling back to (`1`, `1.125`, `1.75`, `2.25`, `3`) don't mean much to users compared to the t-shirt sizes. 

That's all to say that yes, the current approach with t-shirt sizes sound good to me. 👍

**Comment by ntsekouras** [comment]
>That's all to say that yes, the current approach with t-shirt sizes sound good to me. 👍

@kjellr with this approach the alternative  changes for twenty twenty two would be to rename the font sizes to `Small, Medium, Large, Extra Large, Extra Extra Large` and replace all occurrences in the theme where this slugs were used.

This way the `T-shirt` aliases would much the font size names when shown on tooltip, etc..

**Comment by kjellr** [comment]
Ok, makes sense. Just the names, right? The slugs are staying the same? We took the names + slugs directly from core. It gets a little confusing because the sizes named "Medium" & "Large" won't correspond to the `medium` and `large` slugs...

- Small ➡️ Small
- Normal ➡️ Medium
- Medium ➡️ Large
- Large ➡️ Extra Large
- Huge ➡️ Extra Extra Large

**Comment by ntsekouras** [comment]
>Just the names, right? The slugs are staying the same? 

No, the slugs should be updated too so as not create any confusion as you also mention.

**Comment by kjellr** [comment]
Ok, got it. And just confirming that these default slugs are changing in core too? My concern is just that folks with existing content might suddenly have different font sizes since the slugs don't sign to the old core slugs (`normal` doesn't exist, `medium` is now `large`, etc.).

**Comment by ntsekouras** [comment]
>My concern is just that folks with existing content might suddenly have different font sizes since the slugs don't sign to the old core slugs (normal doesn't exist, medium is now large, etc.).

Oh.. that's true..

I'm not sure if that would be the best approach then.. 🤔

**Comment by jasmussen** [comment]
For what it's worth, when I tested this PR, the segmented L → XXL control worked without changing any labels for TT2. While I think it'd be smart for TT2 to match up the two so the tooltip reads better (especially M>Normal), it doesn't seem like it needs to be a hard rule?

**Comment by ntsekouras** [comment]
>For what it's worth, when I tested this PR, the segmented L → XXL control worked without changing any labels for TT2.

It does work but there is the label mismatch like `M->normal`, `L->medium`.

**Comment by ntsekouras** [comment]
@kjellr  has the theme been released independently from 5.9? I'm not familiar with the exact process here..

**Comment by kjellr** [comment]
> @kjellr has the theme been released independently from 5.9? I'm not familiar with the exact process here..

No, the theme won't be officially released until 5.9 is released. Then it'll be bundled with Core and also packaged up separately for the Theme Directory download. 

We've been [tagging some pre-release versions](https://github.com/WordPress/twentytwentytwo/tags), but those are mostly just to make testing easier.  They're not official releases. 

> My concern is just that folks with existing content might suddenly have different font sizes since the slugs don't sign to the old core slugs (normal doesn't exist, medium is now large, etc.).

I dug into this a little more, so to clarify:

When a theme doesn’t provide font size presets, Gutenberg provides some by default ([I think those are defined here?](https://github.com/WordPress/gutenberg/blob/61de10636cca2869269db2c20315b763d11927fc/lib/compat/wordpress-5.9/theme.json#L196-L221)). Twenty Twenty-Two is just copying the names + slugs used there. 

Looks like this PR doesn’t change those default values, which is good! Lots of users have content that uses these default slugs. The default names there don't line up correctly with the t-shirt sizes, but for the default values we don't _really_ have to worry about it because they show up as pixel values: 
 
Screen Shot 2021-12-03 at 2 32 26 PM

So bottom line: We can change this in Twenty Twenty-Two if we need to since it hasn't been released. But conceptually, it's a little weird that core provides a t-shirt scale for font sizes that isn't in sync with the naming used by its own defaults... but it's not a huge problem since users won't really know about it.

**Comment by ntsekouras** [comment]
Any thoughts about @kjellr [comment above](https://github.com/WordPress/gutenberg/pull/37038#issuecomment-985757099) @youknowriad @mtias @jorgefilipecosta @oandregal ?

What are the results of having different font sizes(names,slugs) for twenty twenty two and the default values in `theme.json`. My instinct says that there shouldn't be any overlaps, but is this the case? 🤔  I mean, it's not something that every theme does, so how is this different? Is it because it will be the default `core` theme?

How changing the slugs in theme.json defaults can affect things? Since it's under `compat/5.9` is it something we can change now there as well? When the above default values are used and can we have something like a migration?

Sorry for all these questions, but It seems quite a problem and I'm not 100% sure of the implications this might cause.

**Comment by oandregal** [comment]
Hey, I'm checking the current direction and this is what I understand:

- The idea is to show T-shirt sizes when the theme provides complex CSS values so we can keep using the segmented control. 
- When using the segmented control, the tooltips are the only place in which we show the theme font size names.

What's under discussion is whether the theme slugs/name should or shouldn't use the T-shirt scale, correct?

I see how using the t-shirt scale is inconvenient for how current themes operate: some want to use the same slugs as core to cover the cases of patterns and old content created with core's font sizes. If we wanted to align theme font size names with this scale, we'd also need to rename the core values. The font sizes slugs/names have been like that prior to `theme.json` (probably since 5.0?). Removing any of core's slug is doable but also a sensible thing to do: we'll still need to enqueue the old slugs in CSS for backward compatibility but hide them in the UI.

I align with Joen's [thoughts](https://github.com/WordPress/gutenberg/pull/37038#issuecomment-985681829). It doesn't seem a requirement to align the theme font size names/slugs with the T-shirt scale for 5.9.

**Comment by ntsekouras** [comment]
Thanks for the feedback @oandregal!

>When using the segmented control, the tooltips are the only place in which we show the theme font size names.

Just noting that the selected name is also shown next to the `size` label as well, but it's the same issue 😄 

From your [above comment](https://github.com/WordPress/gutenberg/pull/37038#issuecomment-987817984) I understand that changing slugs/names in core font sizes should be considered, is doable, but doesn't need to happen right now, even if this PR lands which will lead to the labels mismatch. Correct?


If there is consensus about this, the PR is ready to land - has been reviewed, is tested and approved.

**Comment by ntsekouras** [comment]
>The font sizes slugs/names have been like that prior to theme.json (probably since 5.0?). Removing any of core's slug is doable but also a sensible thing to do: we'll still need to enqueue the old slugs in CSS for backward compatibility but hide them in the UI.

@oandregal is this something we should consider independently from this PR for better semantics or any other reason?
I'm a bit hesitant of making these changes just to accommodate our use case for twenty twenty two and core, because the label mismatch can happen with any other theme that meets the conditions to display `T-shirt` aliases.

Should we reconsider the first iteration of this PR, which used an extra `alias` property in font sizes? I'm not very fond of that either as it can introduce some confusion about when the `alias` property is used or not, but it's definitely more explicit.

**Comment by noisysocks** [comment]
What's the verdict on this? Is it a blocker for 5.9? cc. @youknowriad @mtias 

Beta 4 is our last chance to include it. @tellthemachines will be packaging that up on Monday 20 December.

**Comment by jasmussen** [comment]
I'll need to defer to the leads, but here are reasons why I'd be in favor:

- It's a better TwentyTwentyTwo experience
- It does not add an additional API and we can change the labels if need be in the future

For clarity and to help decision making, there are challenges too:

- The t-shirt labels optimizes for 5 sizes, S to XXL. If a theme defines 3 presets that might logically map to M, L and XL, if named in sequence we'd get S/M/L.
- The above is a reason why number in sequence, 1-5, might be better.

One of the higher level concerns is that the slugs carry over with theme switches. The default slugs are `small`, `normal` (personal pet peeve, what is "normal"), `medium`, `large`, `huge`, which means the markup of a theme without font sizes will use these slugs. TT2 maps to that. In addition to that, themes can define their own custom slugs, which expands the range of options to support. Outside of revisiting those default slugs, there's an argument to make that the numbers catch a wider range of options. For better and worse, the numbers just go from small to larger, and need to be (and are) paired with the labels the theme defines.

**Comment by ntsekouras** [comment]
>The above is a reason why number in sequence, 1-5, might be better.

>Outside of revisiting those default slugs, there's an argument to make that the numbers catch a wider range of options. For better and worse, the numbers just go from small to larger, and need to be (and are) paired with the labels the theme defines.

Thanks for the input @jasmussen ! I like your idea and IMO is definitely better from the current iteration of `t-shirt` sizes. This way indeed we don't add a bit complex to comprehend API(`alias`) due to conditional usage, although it's still an option.

I just pushed the changes to use the incremental sequence of numbers as labels for better testing.


---
I'll have to update the PR's title and description when we reach a consensus 😄

**Comment by mtias** [comment]
Gathering all the feedback so far and drawing some points:

- We don't want a complex API; the aim is to reduce overhead as much as possible and increase consistency.
- We want to be able to present most cases with the toggle control and reduced options (2-5 range).
- There should be some level of equivalence when switching themes through standard mappings.

With that said, this is what I think we should aim for now:

Frame 944

We supply some default values (as we do today) for four presets. Themes can customize them just by referencing the slug / type.

```json
"fontSizes": [
	{
		"type": "small",
		"size": "13px"
	},
	{
		"type": "medium",
		"size": "20px"
	},
	{
		"type": "large",
		"size": "36px"
	},
	{
		"type": "x-large",
		"size": "42px"
	}
],
```

- Let's skip the need for themes to provide any sort of name or alias if they want to support the common array.
- We start with `small`, `medium`, `large`, `x-large` slugs that themes can supply presets for. (We still support previous enqueues as needed, nothing should break.)
- When we cannot display the real value we default to 1, 2, 3, 4 — but we might change this to t-shirt sizes at any point. It doesn't affect how themes set things up at all since it's an internal mapping.
- Display labels and short aliases are all internal mappings for this set.
- We leave room for a `xx-large` if needed.
- Themes can add new ones beyond 5 and they get the dropdown selector.

**Comment by ntsekouras** [comment]
>With that said, this is what I think we should aim for now:

> Frame 944

This is the current state of this PR without needing any other changes.

>We supply some default values (as we do today) for four presets. Themes can customize them just by referencing the slug / type.

@mtias  If I understand correctly, the second part of your comments is related to @oandregal [comment](https://github.com/WordPress/gutenberg/pull/37038#issuecomment-987817984) about changing the current `core` font sizes' names and slugs. IMO this is something that is not blocking this PR and I don't know if it's something we should aim for 5.9, so I'd like your opinion on this.

I think we can create a separate issue for the font sizes changes -  I'd appreciate it so much if @oandregal or @kjellr did that for explaining the issue in the best possible way.

**Comment by kjellr** [comment]
> I think we can create a separate issue for the font sizes changes - I'd appreciate it so much if @oandregal or @kjellr did that for explaining the issue in the best possible way.

Sure, I'll open that issue today.

_Edit:_ Done here: https://github.com/WordPress/gutenberg/issues/37378

**Comment by oandregal** [comment]
Got an initial PR to update the default font sizes at https://github.com/WordPress/gutenberg/pull/37381 I'd appreciate heavy testing to make sure that by doing this, we don't introduce any regressions for normal & huge classes/variables.

Edit: also, I presume we want to update the default font-sizes in 5.9?

**Comment by ntsekouras** [comment]
I just rebased and with a  green CI, I'm going to merge this. The next PR to be merged is the one that changes [the core presets](https://github.com/WordPress/gutenberg/pull/37381#pullrequestreview-832432813) and that will only leaves a PR that I'll start today for the below:

> We supply some default values (as we do today) for four presets. Themes can customize them just by referencing the slug / type.
>- Let's skip the need for themes to provide any sort of name or alias if they want to support the common array.
> - We start with `small`, `medium`, `large`, `x-large` slugs that themes can supply presets for. (We still support previous enqueues as needed, nothing should break.)

Noting that we will also need the PR from @kjellr to [change TwentyTwentyTwo](https://github.com/WordPress/twentytwentytwo/pull/298)

**Comment by mtias** [comment]
Thanks for all the work here, @ntsekouras !

**Comment by scruffian** [comment]
> Themes can add new ones beyond 5 and they get the dropdown selector.

The problem with this is that if themes don't follow a standard naming convention, when users switch themes, the additional font-sizes will break. I think it would be good to standardize on a naming convention all themes can use so that if themes do want to add more they will be persisted across themes.

**Comment by ntsekouras** [comment]
>The problem with this is that if themes don't follow a standard naming convention, when users switch themes, the additional font-sizes will break.

Wasn't  this the case before too? I guess it's the same thing with theme's presets of colors, etc.., no?

**Comment by scruffian** [comment]
Yes, it was the case before this change and is also an issue with colors.

**Pull request merged**
